### PR TITLE
Improve test parallelizability

### PR DESCRIPTION
**Pull request by elprans**
We have a bunch of test suites currently that aren't parallelized. Reason number one is absense of proper error recovery handling, which is easily fixable by proper application of `assertRaisesRegexTx`. Reason number two is DDL concurrency and possible deadlocking, which is fixed by making copies of test databases for each runner worker.  Altogether, this meaningfully improves test run times on machines with high core count, but is also useful to speed up selective test running if the selected test suite happens to be non-parallelizable currently.

### CODE REVIEW

File: edb/testbase/server.py [hunk 1]
```diff
@@ -643,8 +644,17 @@ class DatabaseTestCase(ClusterTestCase, ConnectedTestCaseMixin):
...
-    ISOLATED_METHODS = True
+    # or affect non-transactional state, in which case
+    # TRANSACTION_ISOLATION must be set to False
+    TRANSACTION_ISOLATION = True
+
+    # By default, tests from the same testsuite may be ran
+    # in parallel in several test worker processes.  However,
+    # certain cases might exhibit pathological locking behavior,
+    # or are parallel-unsafe altogether, in which case
+    # PARALLELISM_GRANULARITY must be set to 'database' or 'system'.
```
Comment (1st1) [inline edb/testbase/server.py:657]: This comment doesn't actually explain what "database" or "system" do.

File: tests/test_cli.py [hunk 2]
```diff
@@ -1,72 +0,0 @@
...
-# limitations under the License.
-#
-
-
-import edgedb
-
-from edb.testbase import server as tb
-
-
-class TestCLI(tb.ConnectedTestCase, tb.OldCLITestCaseMixin):
```
Comment (1st1) [inline tests/test_cli.py:25]: Is this removal intentional? The commit message doesn't mention it.

File: edb/edgeql/parser/grammar/ddl.py [hunk 3]
```diff
@@ -424,12 +424,22 @@ def reduce_AlterFinal(self, *kids):
 # CREATE DATABASE
 #
 class CreateDatabaseStmt(Nonterm):
-    def reduce_CREATE_DATABASE_AnyNodeName_OptCreateRoleCommandsBlock(
-        self,
-        *kids,
-    ):
+    def reduce_CREATE_DATABASE_regular(self, *kids):
+        """%reduce CREATE DATABASE AnyNodeName OptCreateDatabaseCommandsBlock
```
Comment (1st1) [inline edb/edgeql/parser/grammar/ddl.py:428]: And this commit actually adds a new feature to EdgeQL that needs to be documented and perhaps deserves a commit message.

Comment (elprans) [reply edb/edgeql/parser/grammar/ddl.py:428]: It's a hidden feature, only for `testmode`.  There is no intention to expose it for now.

File: tests/test_cli.py (hunk 2 above)
Comment (elprans) [reply tests/test_cli.py:25]: The old CLI is slated for removal and I couldn't be bothered to spend time to fix something that's going to get removed very soon anyway.

File: edb/testbase/server.py (hunk 1 above)
Comment (elprans) [reply edb/testbase/server.py:657]: Added an explanation.

File: edb/edgeql/parser/grammar/ddl.py (hunk 3 above)
Comment (1st1) [reply edb/edgeql/parser/grammar/ddl.py:428]: Ah, I see you raise an error in `_cmd_tree_from_ast`. All's good.

**Comment by elprans** [comment]
I hid the parallel-mode behind a debug flag, so this should be safe to merge.

**Review by 1st1 — APPROVED** [review]
